Updated OverlayFS for RedHat 7 (CVM-835)

radupopescu opened the pull request:
This branch:
* [x] Updates the function `check_overlayfs_version` in `cvmfs_server`. The new success condition is either kernel 4.2.0 or newer on all Linux distributions, or, if running RedHat, kernel 3.10.0-493 or newer AND /var/spool/cvmfs should be located on an ext4 partition.
* [x] Updates the integration test scripts to run the server tests on RedHat (CentOS) 7, setting up `/var/spool/cvmfs` on an ext4 partition.

jblomer commented on test/cloud_testing/platforms/slc6_x86_64_test.sh:
```diff
@@ -73,12 +73,68 @@ CVMFS_TEST_CLASS_NAME=ServerIntegrationTests                                  \
                               || retval=1
 
 
+echo -n "starting FakeS3 service... "
```
The S3 tests unfortunately need to be disabled for the time being.  They all fail, I think because of FakeS3.  But we didn't look into it yet.

radupopescu replied:
Alright. I will go through all the test scripts and disable the S3 tests, to reduce the noise in the test output.

jblomer commented:
Let's check if the overlay-fs validation test works on Ubuntu before merging.  Otherwise we'd activate it only on Fedora and CentOS7.

radupopescu commented:
I think I may have made a mistake in the last commit. I need to double-check

radupopescu commented:
I reverted the previous commit. The test no longer hangs and I added more detailed status reporting.

It seems that the first test case, trying to delete files in the lower ro branch of the union fs is failing.

I will run the integration test on CentOS 7, too.

Jakob, could you please check the file `src/700-overlayfsvalidation/main`?

radupopescu commented:
I ran the test on CentOS 7. The same test case is failing as on Ubuntu.

radupopescu commented:
I've triggered another run of the integration tests (5** and 7**) in my CentOS 7 VM with Ext4 scratch. Will post here the results when they're done.

jblomer commented:
@radupopescu I guess this is ready, or do you want to add anything for xfs with fstype=1 to this PR?

radupopescu commented:
I'd like to try the OverlayFS tests on a volume with `ftype=1` and then we can close.
